generate_schedule.py

```diff
@@ -13,25 +13,20 @@
 EPS = 1e-8
 
 
-def generate_flex_band(scenario, core_standing_time=None):
+def generate_flex_band(scenario, gc_index, core_standing_time=None):
     """Generate flexibility potential with perfect foresight
-
     :param scenario: dictionary from scenario json
     :type scenario: dict
     :param core_standing_time: core standing time during which flexibility is guaranteed e.g.
         {"times":[{"start": [22,0], "end":[5,0]}], "full_days":[7]}
     :type core_standing_time: dict
     :return: flex band
     :rtype: dict
-
     """
-
-    assert len(scenario.constants.grid_connectors) == 1, "Only one grid connector supported"
-    gc = list(scenario.constants.grid_connectors.values())[0]
-
+    gc = list(scenario.constants.grid_connectors.values())[gc_index]
     # generate basic strategy
     s = strategy.Strategy(
-        scenario.constants, scenario.start_time, **{"interval": scenario.interval})
+        scenario.constants, scenario.start_time, **{"interval": scenario.interval, "margin": 1})
```

Comment:
margin = 1 solves the problem if cars leave at the same timestep they are arriving. It does not solve the different problem of not making the next trip because the SoC is too low (e.g., PHEV). Three options:

1. ignore (simulation may fail in post-processing)
2. include allow_negative_soc
3. instead of using margin=1, use a try...catch block around s.step(...) -> this would also catch all other simulation errors